distsys/archetyperesource.go

```diff
@@ -50,6 +56,10 @@ type ArchetypeResourceState interface {
 	// ErrCriticalSectionAborted.
 	// This makes no sense for a value-like resource, and should be blocked off with ArchetypeResourceLeafMixin in that case.
 	Index(index tla.TLAValue) (ArchetypeResourceComponent, error)
+	// ForkState provides a copy to the ArchetypeResourceState such that operations performed on other ArchetypeResourceStates
+	// with the same parent ArchetypeResource will not cause side effects for the current ArchetypeResourceState.
+	// ForkState will clone all the parts of a resource whose properties are NOT idempotent. A call to ForkState
+	// creates the expectation that there will be an eventual call to Commit/Abort on the resulting ArchetypeResourceState
```

Comment:
Suggested rephrasing:

~~ForkState will clone all the parts of a resource whose properties are NOT idempotent.~~

ForkState should create a separate copy of any thread-local data held by the original ArcvhetypeResourceState. It is the responsibility of the implementation to ensure side-effects that are not thread-local are synchronized and maintained correctly, given that the two forked versions will be accessed by different goroutines.